### PR TITLE
Pin Pylint to 2.4.4

### PR DESCRIPTION
**Pull request by kaxil**
Pylint 2.5.0 was released 8 hours ago ([link](https://pypi.org/project/pylint/2.5.0/)) which has added bunch of new rules & checks.

Pinning to 2.4.4 till we check the new rules & disable the ones we don't want to use

---
Make sure to mark the boxes below before creating PR: [x]

- [x] Description above provides context of the change
- [x] Unit tests coverage for changes (not needed for documentation changes)
- [x] Target Github ISSUE in description if exists
- [x] Commits follow "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)"
- [x] Relevant documentation is updated including usage instructions.
- [x] I will engage committers as explained in [Contribution Workflow Example](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#contribution-workflow-example).

---
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/master/UPDATING.md).
Read the [Pull Request Guidelines](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#pull-request-guidelines) for more information.

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -444,7 +444,7 @@ def write_version(filename: str = os.path.join(*[my_dir, "airflow", "git_version
     'parameterized',
     'paramiko',
     'pre-commit',
-    'pylint~=2.4',
+    'pylint==2.4.4',
```
Comment (feluelle) [inline setup.py:447]:
```suggestion
    'pylint~=2.4,<2.5.0',
```

**Review by feluelle — APPROVED** [review]
LGTM

File: setup.py (hunk 1 above)
Comment (kaxil) [reply setup.py:447]: pylint~=2.4 -> `pylint>=2.4,==2.*` so you can't add <2.5.0 with it :)

Comment (feluelle) [reply setup.py:447]: Ah okay - get it 👍

**Comment by potiuk** [comment]
Nice :)

**Pull request merged**